app/cdap/components/GridTable/index.tsx

```diff
@@ -170,20 +174,24 @@ export default function GridTable() {
   // ------------@getGridTableData Function is used for preparing data for entire grid-table
   const getGridTableData = async () => {
     const rawData: IExecuteAPIResponse = gridData;
-    const headersData = createHeadersData(rawData.headers, rawData.types);
-    setheaderNamesList(headersData);
+    const headersData: IHeaderNamesList[] = createHeadersData(
+      rawData?.headers,
+      rawData?.types
+    ) as IHeaderNamesList[];
+    setHeaderNamesList(headersData);
     if (rawData && rawData?.summary && rawData?.summary?.statistics) {
-      const missingData = createMissingData(gridData?.summary?.statistics);
-      setMissingDataList(missingData);
+      const missingItems: IMissingList[] = getMissingList(gridData?.summary?.statistics);
+      setMissingDataList(missingItems);
+      setDataQuality(gridData?.summary?.statistics);
     }
-    const rowData =
+    const rowData: ICellData[] =
       rawData &&
-      rawData.values &&
+      rawData?.values &&
       Array.isArray(rawData?.values) &&
-      rawData?.values.map((eachRow: {}) => {
+      (rawData?.values?.map((eachRow: {}) => {
         const { ...rest } = eachRow;
         return rest;
-      });
+      }) as ICellData[]);
```

Comment:
we should try not to use as ABC; for typing issues
plz try to resolve by adding proper interface or types